judge/models/contest.py

```diff
@@ -86,6 +86,13 @@ class Contest(models.Model):
     problems = models.ManyToManyField(Problem, verbose_name=_('problems'), through='ContestProblem')
     start_time = models.DateTimeField(verbose_name=_('start time'), db_index=True)
     end_time = models.DateTimeField(verbose_name=_('end time'), db_index=True)
+    registration_start = models.DateTimeField(verbose_name=_('registration start time'),
+                                              blank=True, null=True, default=None)
+    registration_end = models.DateTimeField(verbose_name=_('registration end time'),
+                                            blank=True, null=True, default=None)
+    registered_users = models.ManyToManyField(Profile, help_text=_('Users who registered for the contest.'),
```

Comment:
on registration, we will create a `ContestParticipation` object, we can use that to check for registered users instead of creating this many-to-many field.